api-standards/structure.md

```diff
@@ -0,0 +1,84 @@
+An extension of the [API Standards](../API_STANDARDS.md), with implementation details specific to development of RESTful APIs for ONS.
+
+## RESTful API Structure and Syntax
+
+### Syntax & types
+
+  * Field names snake case, e.g. `total_count`
+  * Use appropriate types for fields, e.g. numbers should be ints not strings
+  * Timestamps should be UTC (optionally, with a timezone)
+  * Root endpoint IDs should always be referred to as `id`, rather than name-spaced. All IDs under this level will need descriptive placeholder names. e.g. /datasets/<id>/editions/<edition> where /datasets is the root endpoint
+
+### Methods & Behaviours
+ * Use the `GET` method for search endpoints
+ * `POST` endpoints should typically generate IDs, and these should use GUIDs not auto-incrementing counts
+
+ ### Links
+* All endpoints must return a `links.self` subdoc in their response,
+* Resources should avoid a nested list by having a links to list endpoints
+* Responses must contain fully qualified URLs, though this may be managed by the API router
+* `Links` object contains descriptively named objects which each contain
+  an `id` (optional) and `url` field e.g.
+  ```
+  {  
+      "id":"1234",
+      "data":"i have stuff",
+      "links": {  
+          "latest_resource": {  
+              "id":"5678",
+              "url":"/thisapi/1234/subresources/5678"
+          },
+          "external": {  
+              "id":"3456",
+              "url":"/extresources/3456"
+          },
+          "subresources":{
+              "url":"/thisapi/1234/subresources"
+          },
+          "self":{
+              "id":"1234",
+              "url":"/thisapi/1234"
+          }
+      }
+  }
+  ```
+* Resources may link to their direct parent, a child list, or any relevant
+  external resources. Resources which are many layers deep in an API do not need
+  to contain links to all higher elements e.g. the resource at
+  `/datasets/1234/editions/2345/versions/1` would contain a link to
+  `/datasets/1234/editions/2345` but not to `/datasets/1234`
+
+
+### Returning errors
+  * **Errors** should return a status code and a JSON payload with the error message
```

Comment:
should not found be 404 or 400 with an error message ?

Reply:
So not found can be either 400 or 404; 404 is more explicit than 400 for not found and hence we tend to use 404 status codes in our APIs.

Do we think it is important to specify when we use certain status codes for particular mistakes made by the API user here or allow teams more autonomy to decide how specific they would like to get with status code?

Reply:
Not found should be 404, but ill make sure thats clear

Reply:
I actually think it'd be helpful to go into some detail, for instance I was recently explaining the security risk introduced by exposing 403s on unpublished pages rather than obscuring them behind 404s. 

But this PR was meant to primarily be a structural change, I havent actually changed what this section says so I'd prefer that additional detail be a separate PR